### PR TITLE
Shouldn't use tempnam

### PR DESCRIPTION
**Pull request by ksss**
This warning was displayed when building.

```
/Users/ksss/src/github.com/ksss/mruby/mrbgems/mruby-socket/test/sockettest.c:13:15: warning: 'tempnam' is
      deprecated: This function is provided for compatibility reasons only. Due to security concerns
      inherent in the design of tempnam(3), it is highly recommended that you use mkstemp(3) instead.
      [-Wdeprecated-declarations]
  char *tmp = tempnam(NULL, "mruby-socket");
              ^
/usr/include/stdio.h:395:1: note: 'tempnam' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security conc...
^
/usr/include/sys/cdefs.h:180:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((deprecated(_msg)))
```

### CODE REVIEW

**Comment by matz** [comment]
This change does not make the test secure. 

We used `tempnam` to avoid potential file name crash. `tempnam` is not encouraged because it is possible to guess scrambled file name before opening it. Your fix does not scramble at all, that means the attacker does not needs to guess.

We may need to fix the code to remove `tempnam`, but not this way.

**Comment by ksss** [comment]
OK, I fixed to use `mkstemp` like mruby-io/test

**Comment by matz** [comment]
Security-wise, it's not ideal (as bad as `mruby-io` test), but acceptable.

File: mrbgems/mruby-socket/test/sockettest.c [hunk 1]
```diff
@@ -2,18 +2,49 @@
 #include <stdlib.h>
 
 #include "mruby.h"
+#include "mruby/error.h"
+
+#if defined(_WIN32) || defined(_WIN64)
+
+#define close _close
+#define unlink _unlink
```
Comment (mattn) [inline mrbgems/mruby-socket/test/sockettest.c]: please define these only for _MSC_VER.

File: mrbgems/mruby-socket/test/sockettest.c [hunk 2]
```diff
@@ -2,18 +2,49 @@
...
+
+#ifdef _MSC_VER
```
Comment (mattn) [inline mrbgems/mruby-socket/test/sockettest.c]: So please move up this before `#define close _close`.

**Comment by mattn** [comment]
I'll send fix in later.

**Comment by matz** [comment]
I fixed it in f7ff2d0e

**Comment by ksss** [comment]
Thank you for fixing.

**Pull request merged**